src/ui/src/composables/useFilesEncoder/useFilesEncoder.ts

```diff
@@ -0,0 +1,114 @@
+import { MaybeRef, readonly, ref, toValue } from "vue";
+import { encodeFileAsDataURL } from "./encodeFileAsDataURL";
+
+export type UiFile = {
+	id: string;
+	name: string;
+	size: number;
+	file: File;
+};
+
+let fallbackId = 0;
+
+function convertFileToUiFile(file: File): UiFile {
+	const id = window.crypto?.randomUUID() ?? `file-${fallbackId++}`;
+
+	return {
+		id,
+		file,
+		name: file.name,
+		size: file.size,
+	};
+}
+
+export type EncodedFile = {
+	name: string;
+	type: string;
+	data: string;
+};
+
+export type UseFilesEncoderParams = {
+	multiple: MaybeRef<boolean>;
+};
+
+export function useFilesEncoder({ multiple }: UseFilesEncoderParams) {
+	const uiFiles = ref<UiFile[]>([]);
```

Comment:
It's better to use `shallowRef`, it's save a bit of memory

```suggestion
	const uiFiles = shallowRef<UiFile[]>([]);
```